tests/phpunit/tests/rest-api/rest-attachments-controller.php

```diff
@@ -619,6 +619,33 @@ public function test_get_items_valid_modified_date() {
 		$this->assertSame( $id2, $data[0]['id'] );
 	}
 
+	/**
+	 * @ticket 55677
+	 */
+	public function test_get_items_avoid_duplicated_count_query_if_no_items() {
+		$request = new WP_REST_Request( 'GET', '/wp/v2/media' );
+		$request->set_param( 'media_type', 'video' );
+		$initial_num_queries = get_num_queries();
```

Comment:
Tests that verify `num_queries` have the potential to be brittle. Could we update this to use similar logic as the tests you updated? That is checking for the actual `WP_Query`s that are run. For instance by checking the number of time `post_clauses` is executed.

Reply:
Thanks @TimothyBJacobs for the feedback. I've now changed these tests to use `post_clauses` as you suggested

Reply:
Looks great, thanks!